config.go

```diff
@@ -394,9 +408,32 @@ func (cfg *Config) unshadow() {
 	cfg.shadow = nil
 }
 
+func generateSwarmSecret() (string, error) {
+	key := make([]byte, 32)
+	_, err := crand.Read(key)
+	if err != nil {
+		return "", fmt.Errorf("Error reading from rand: %v", err)
+	}
+	return hex.EncodeToString(key), nil
+}
+
+func swarmSecretToKey(secret string) (string, error) {
+	hexSecret := hex.EncodeToString([]byte(secret))[:68]
```

Comment:
I am open to PR to go-libp2p-pnet that adds direct handling of `[]byte` secret, it would simplify this part quite a bit.